zebra/kernel_netlink.c

```diff
@@ -1513,11 +1523,35 @@ void kernel_update_multi(struct dplane_ctx_q *ctx_list)
 
 struct nlsock *kernel_netlink_nlsock_lookup(int sock)
 {
-	struct nlsock lookup;
+	struct nlsock lookup, *retval;
 
 	lookup.sock = sock;
 
-	return hash_lookup(nlsock_hash, &lookup);
+	NLSOCK_LOCK();
+	retval = hash_lookup(nlsock_hash, &lookup);
+	NLSOCK_UNLOCK();
+
+	return retval;
+}
+
+/* Insert nlsock entry into hash */
+static void kernel_netlink_nlsock_insert(struct nlsock *nls)
+{
+	NLSOCK_LOCK();
+
+	(void)hash_get(nlsock_hash, nls, hash_alloc_intern);
+
+	NLSOCK_UNLOCK();
+}
+
+/* Remove nlsock entry from hash */
+static void kernel_netlink_nlsock_release(struct nlsock *nls)
```

Comment:
Because we now have `nlsock_mutex`,  on first sight, I thought `nlsock_release` has to do with releasing the mutex. Can we just call this  `kernel_netlink_nlsock_remove` to avoid terminology mix up   in this case?   The  remove function obviously release the hash, and it match the `insert()` function  you have better.

Reply:
I picked "release" because it matches "hash_release", and I thought that would be helpful. but I'm not attached to it - I'll rename it.